src/main/java/uk/nhs/tis/trainee/notifications/service/PlacementService.java

```diff
@@ -221,4 +257,64 @@ private boolean shouldScheduleNotification(
     //do not resend any notification
     return (!notificationsAlreadySent.containsKey(PLACEMENT_UPDATED_WEEK_12));
   }
+
+  /**
+   * Create any relevant in-app notifications.
+   *
+   * @param placement      The updated placement.
+   * @param notificationsAlreadySent Previously sent notifications.
+   */
+  private void createInAppNotifications(Placement placement,
+                                        Map<NotificationType, Instant> notificationsAlreadySent) {
+    boolean meetsCriteria = notificationService.meetsCriteria(placement, true);
+
+    if (meetsCriteria) {
+      String owner = placement.getOwner();
+      List<Map<String, String>> contactList = notificationService.getOwnerContactList(owner);
+
+      // PLACEMENT_INFORMATION
+      String localOfficeContact = notificationService.getOwnerContact(contactList,
+          LocalOfficeContactType.TSS_SUPPORT, null);
+      String localOfficeContactType =
+          notificationService.getHrefTypeForContact(localOfficeContact);
+      createUniqueInAppNotification(placement, notificationsAlreadySent, PLACEMENT_INFORMATION,
+          placementInfoVersion, Map.of(
+              LOCAL_OFFICE_CONTACT_FIELD, localOfficeContact,
+              LOCAL_OFFICE_CONTACT_TYPE_FIELD, localOfficeContactType));
+    }
+  }
+
+  /**
+   * Create a unique in-app notification of the given type and version.
+   *
+   * @param placement                The updated placement.
+   * @param notificationsAlreadySent Previously sent notifications.
+   * @param notificationType         The type of notification being sent.
+   * @param notificationVersion      The version of the notification.
+   * @param extraVariables           Extra variables to include with the template, Specialty,
+   *                                 Site Known As and Start Date are populated automatically.
+   */
+  private void createUniqueInAppNotification(Placement placement,
+      Map<NotificationType, Instant> notificationsAlreadySent, NotificationType notificationType,
+      String notificationVersion, Map<String, Object> extraVariables) {
+    boolean isUnique = !notificationsAlreadySent.containsKey(notificationType);
```

Comment:
I think we should drop future-dated (sentAt is in the future) placement notifications and rebuild them, not ignore subsequent changes to a placement if we have already set up an in-app notification for it. We do this for the emails. The reason is that if a placement has been changed after it was first created, we should show the latest information when it comes time to display the in-app notification. This will also mean it matches with the placement email that is sent at the same time.